java/java.hints/src/org/netbeans/modules/java/hints/bugs/Unused.java

```diff
@@ -56,12 +62,33 @@ public class Unused {
     @TriggerTreeKind(Kind.COMPILATION_UNIT)
     public static List<ErrorDescription> unused(HintContext ctx) {
         List<UnusedDescription> unused = UnusedDetector.findUnused(ctx.getInfo(), () -> ctx.isCanceled());
+        Set<UnusedDescription> toRemove = new HashSet<>();
+        if (!unused.isEmpty()) {
+            List<TestMethodController.TestMethod> tms = getTestMethods(ctx.getInfo());
```

Comment:
I'd prefer some SPI that other modules could plug into. For example, `@RequestMapping` or other annotations could be put even on a private method (although discouraged), and should be deemed "used" -- tracked as #6284

Reply:
Required SPI created. Implementation changed to used the newly crated SPI.